fe/fe-core/src/main/java/org/apache/doris/analysis/AlterTableStatsStmt.java

```diff
@@ -35,61 +39,125 @@
-                .filter(statsType -> !CONFIGURABLE_PROPERTIES_SET.contains(statsType)).findFirst();
+                .filter(statsType -> !CONFIGURABLE_PROPERTIES_SET.contains(statsType))
+                .findFirst();
         if (optional.isPresent()) {
-            throw new AnalysisException(optional.get() + " is invalid statistic");
+            throw new AnalysisException(optional.get() + " is invalid statistics");
         }
+
         // check auth
-        if (!Env.getCurrentEnv().getAuth().checkTblPriv(
-                ConnectContext.get(), tableName.getDb(), tableName.getTbl(), PrivPredicate.ALTER)) {
+        if (!Env.getCurrentEnv().getAuth()
+                .checkTblPriv(ConnectContext.get(), tableName.getDb(), tableName.getTbl(), PrivPredicate.ALTER)) {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLEACCESS_DENIED_ERROR, "ALTER TABLE STATS",
-                    ConnectContext.get().getQualifiedUser(),
-                    ConnectContext.get().getRemoteIP(),
+                    ConnectContext.get().getQualifiedUser(), ConnectContext.get().getRemoteIP(),
                     tableName.getDb() + ": " + tableName.getTbl());
         }
+
         // get statsTypeToValue
         properties.forEach((key, value) -> {
             StatsType statsType = StatsType.fromString(key);
             statsTypeToValue.put(statsType, value);
         });
     }
 
-    public TableName getTableName() {
-        return tableName;
-    }
+    private void checkPartitionNames() throws AnalysisException {
+        if (optPartitionNames != null) {
+            optPartitionNames.analyze(analyzer);
+            Database db = analyzer.getEnv().getInternalDataSource()
+                    .getDbOrAnalysisException(tableName.getDb());
+            Table table = db.getTableOrAnalysisException(tableName.getTbl());
 
-    public Map<StatsType, String> getStatsTypeToValue() {
-        return statsTypeToValue;
+            if (table.getType() != Table.TableType.OLAP) {
+                throw new AnalysisException("Only OLAP table statistics are supported");
+            }
+
+            OlapTable olapTable = (OlapTable) table;
+
+            if (olapTable.isPartitioned()) {
+                List<String> names = optPartitionNames.getPartitionNames();
+                Set<String> olapPartitionNames = olapTable.getPartitionNames();
+                Optional<String> optional = names.stream()
+                        .filter(name -> !olapPartitionNames.contains(name)).findFirst();
+                if (optional.isPresent()) {
+                    throw new AnalysisException("Partition does not exist: " + optional.get());
+                }
+                partitionNames.addAll(optPartitionNames.getPartitionNames());
+            } else {
+                throw new AnalysisException("Not a partitioned table: " + olapTable.getName());
+            }
+        }
```

Comment:
should we need to handle table is unpartitioned and optPartitionNames is not empty just like AlterColumnStats?

Reply:
sure

Reply:
There are table level and partition level statistics. For a partitioned table, if a partition is specified by the user, the statistics of the corresponding partition will be updated, and if no partition is specified, the statistics of the table level will be updated.